test/e2e/testdata/authorization-client-ip.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: http-with-authorization-client-ip-1
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: same-namespace
+  rules:
+  - matches:
+    - path:
+        type: Exact
+        value: /protected1
+    backendRefs:
+    - name: infra-backend-v1
+      port: 8080
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: http-with-authorization-client-ip-2
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: same-namespace
+  rules:
+  - matches:
+    - path:
+        type: Exact
+        value: /protected2
+    backendRefs:
+    - name: infra-backend-v1
+      port: 8080
+---
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: SecurityPolicy
+metadata:
+  name: authorization-client-ip-1
+  namespace: gateway-conformance-infra
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: http-with-authorization-client-ip-1
+  authorization:
+    defaultAction: Allow
+    rules:
+    - name: "deny-location-1"  # First matching rule is applied, so 192.168.1.0/24 will be denied
+      action: Deny
+      principal:
+        clientCIDRs:
+        - 192.168.1.0/24
+    - name: "allow-location-1"
+      action: Allow
+      principal:
+        clientCIDRs:
+        - 192.168.1.0/24
+        - 192.168.2.0/24     # First matching rule is applied, so 12.168.2.0/24 will be allowed
+    - name: "deny-location-2"
+      action: Allow
+      principal:
+        clientCIDRs:
+        - 192.168.2.0/24
+---
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: SecurityPolicy
+metadata:
+  name: authorization-client-ip-2
+  namespace: gateway-conformance-infra
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: http-with-authorization-client-ip-2
+  authorization:
+    defaultAction: Deny
+    rules:
+    - action: Allow
+      principal:
+        clientCIDRs:
+        - 10.0.1.0/24
+        - 10.0.2.0/24
+---
+# This is a client traffic policy that enables client IP detection using a custom header.
+# So, the client IP can be detected from the custom header `client-ip` and used for authorization.
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: ClientTrafficPolicy
+metadata:
+  name: enable-client-ip-detection
+  namespace: gateway-conformance-infra
+spec:
+  clientIPDetection:
+    customHeader:
```

Comment:
The current implementation works for the `customHeader` ClientIPDetection, but not for XFF.

In order to also support XFF, we need to use the [XFF IP detection extension](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/original_ip_detection/xff/v3/xff.proto#extension-envoy-http-original-ip-detection-xff) instead of the [xff_num_trusted_hops](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-for) HCM setting for client IP detection.

@davidalger @ardikabs  Are there specific reasons why we prefer `xff_num_trusted_hops` over XFF IP detection extension in the implementation of https://github.com/envoyproxy/gateway/pull/2535 ? Would it be possible to switch to the XFF IP detection extension?